### PR TITLE
feat: Add support for multi-query panels with refId-based file naming

### PR DESCRIPTION
**Pull request by zhehlovvalentyn**
## 🎯 Overview

This PR adds comprehensive support for Grafana panels with multiple queries, implementing a smart file naming convention and sync functionality that maintains backward compatibility.

### �� Multi-Query Export Support
- **Single target panels**: Clean filenames without refId suffix (e.g., `panel1.sql`)
- **Multi target panels**: Unique filenames with refId suffix (e.g., `panel1_a.sql`, `panel1_b.sql`)
- **Intelligent naming**: Automatically detects panel type and applies appropriate naming

### 🔄 Multi-Query Sync Support
- **RefId-based matching**: Updates specific targets based on their `refId`
- **Backward compatibility**: Works with existing single-query panels
- **Smart lookup**: `QueryManager.GetByBaseAndRefId()` for intelligent query resolution

### Core Implementation
- **`pkg/command/client.go`**: 
  - `ExportPanelQueries()`: Multi-query export with conditional refId naming
  - `UpdatePanelTargets()`: RefId-based target updates
  - Made methods public for better testability

- **`pkg/command/query.go`**: 
  - `GetByBaseAndRefId()`: New method for intelligent query lookup
  - Supports both single-target (no suffix) and multi-target (with suffix) scenarios

### Testing
- **`pkg/command/dashboard_export_queries_test.go`**: 
  - Refactored into focused single/multi target tests
  - Removed redundant `TestExportPanelQueriesDuplicateDescriptions`
  - Updated test expectations for new naming convention

- **`pkg/command/dashboard_sync_test.go`**: 
  - New comprehensive sync tests
  - Multi-query, single-query, and backward compatibility scenarios

### File Naming Convention
```go
// Single target (no refId suffix)
panel1.sql
panel2.promql

// Multi target (with refId suffix)
panel1_a.sql
panel1_b.promql
panel1_c.sql
```

### Query Lookup Logic
```go
// GetByBaseAndRefId intelligently searches:
// 1. Base name without suffix (for single targets)
// 2. Base name with lowercase refId suffix (for multi targets)
query := queryManager.GetByBaseAndRefId("panel1", "A")
// Looks for: panel1.sql OR panel1_a.sql OR panel1_a.promql
```

## ✅ Backward Compatibility

- **Existing single-query panels**: Continue to work without changes
- **Old export format**: Still supported and correctly synced
- **Panel descriptions**: No breaking changes to existing workflows

## 🧪 Test Results

All tests pass with comprehensive coverage:
```bash
go test ./pkg/command -v
# 13 tests passing, covering:
# - Export functionality (single/multi target)
# - Sync functionality (single/multi target)  
# - Backward compatibility
# - Edge cases and error handling
```

### Example 1: Single Query Panel
```json
{
  "targets": [{"expr": "up{job=\"test\"}", "refId": "A"}],
  "description": "cpu_usage"
}
// Exports to: queries/cpu_usage.promql
```

### Example 2: Multi Query Panel
```json
{
  "targets": [
    {"expr": "query1", "refId": "A"},
    {"expr": "query2", "refId": "B"}
  ],
  "description": "cpu_usage"
}
// Exports to: queries/cpu_usage_a.promql, queries/cpu_usage_b.promql
```

## �� Review Checklist

- [x] Code follows project conventions
- [x] All tests pass
- [x] Backward compatibility maintained
- [x] Documentation updated
- [x] No breaking changes introduced

---

**Ready for review!** 🚀

### CODE REVIEW

File: pkg/command/client.go [hunk 1]
```diff
@@ -370,25 +382,45 @@ func (c *Client) exportPanelQueries(panel *simplejson.Json, queriesDir string, o
 		return nil
 	}
 
-	// Parse panel description to get query file paths
+	// Parse panel description to get base query path
 	queryPaths := c.parseQueryPaths(panelDesc)
-	if len(queryPaths) == 0 {
-		c.logd("no valid query paths found for panel %s:%q (description: %q)", panelType, panelTitle, panelDesc)
-		return nil
+	var baseQueryPath string
```
Comment (diogogmt) [inline pkg/command/client.go]: I think the same code block is reused in 4 different functions. Can we create a helper function to return the **baseQueryPath** from a **panelDesc**? Maybe we can replace the **pareQueryPaths** since we only reference the **queryPaths** to define the **baseQueryPath**

**Comment by diogogmt** [comment]
> Single target panels: Clean filenames without refId suffix (e.g., panel1.sql)
> Multi target panels: Unique filenames with refId suffix (e.g., panel1_a.sql, panel1_b.sql)

So by default, if we detect multiple targets in the panel, we name the queries by adding a suffix using a lowercase alphabet increment? What happens if a second target was added after the panel was connected to a single query?

> RefId-based matching: Updates specific targets based on their refId

What is a target refID?

**Review by diogogmt — APPROVED** [review]

**Comment by zhehlovvalentyn** [comment]
> > Single target panels: Clean filenames without refId suffix (e.g., panel1.sql)
> > Multi target panels: Unique filenames with refId suffix (e.g., panel1_a.sql, panel1_b.sql)
> 
> So by default, if we detect multiple targets in the panel, we name the queries by adding a suffix using a lowercase alphabet increment? What happens if a second target was added after the panel was connected to a single query?
> 
> > RefId-based matching: Updates specific targets based on their refId
> 
> What is a target refID?

Not exactly. We use refID value as a sufix
for example here
Screenshot 2025-07-22 at 12 41 46 PM
the refIds are `Before October 2024` and `After October 2024`.
The alphabetik increment suffix comes from the default behaviour of Grafana. When you add a new target (query) by default Grafana use alphabetic increment as a name for this target 
e.g.
Screenshot 2025-07-22 at 12 45 35 PM
you can rename it and the appropriate suffix will be used 

from the JSON perspective it looks like this 
```json
{
  "id": 110,
  "type": "timeseries",
  "title": "CPU",
  "description": "query=service-explorer/fingerprinter-explorer/cpu-template/graph-cpu",
  "gridPos": {
    "x": 12,
    "y": 37,
    "h": 10,
    "w": 12
  },
  "fieldConfig": {
  },
  "pluginVersion": "12.0.1",
  "targets": [
    {
      "editorMode": "code",
      "exemplar": true,
      "expr": "avg by (mode)(irate(node_cpu_seconds_total{mode='idle',service=\"$service\",template=\"$template\"})) * 100",
      "format": "time_series",
      "interval": "",
      "intervalFactor": 2,
      "legendFormat": "Idle",
      "range": true,
      "refId": "F",
      "step": 240
    },
    {
      "editorMode": "code",
      "exemplar": true,
      "expr": "avg (irate(node_cpu_seconds_total{mode='user',service=\"$service\",template=\"$template\"})) * 100",
      "format": "time_series",
      "interval": "",
      "intervalFactor": 2,
      "legendFormat": "Busy User",
      "range": true,
      "refId": "B",
      "step": 240
    },
```

File: pkg/command/client.go (hunk 1 above)
Comment (zhehlovvalentyn) [reply pkg/command/client.go]: I think we can do this
